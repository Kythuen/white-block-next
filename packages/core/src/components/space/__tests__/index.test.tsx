import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import Space from '../space.vue'
// import { useSpace } from '../use'

describe('# Space', () => {
  describe('## Base', () => {
    it('### default config', () => {
      const wrapper = mount(Space)
      expect(wrapper.classes()).toEqual([
        'wb-space',
        'wb-space--start',
        'wb-space--md'
      ])
      expect(wrapper.props()).toEqual({
        vertical: false,
        align: 'start',
        gap: 'md'
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('## Slots', () => {
    describe('### content', () => {
      describe('### default', () => {
        it('#### slot: default', () => {
          const wrapper = mount({
            render() {
              return (
                <Space>
                  <button>button1</button>
                  <button>button2</button>
                </Space>
              )
            }
          })
          expect(wrapper.findAll('button').length).toBe(2)
        })
        it('#### prop: default', () => {
          const wrapper = mount(Space, {
            props: {
              default: (
                <>
                  <button>button1</button>
                  <button>button2</button>
                </>
              )
            }
          })
          expect(wrapper.findAll('button').length).toBe(2)
        })
      })
    })
  })

  // describe('## Events', () => {})

  // describe('## Props', () => {})

  // describe('## Composables', () => {})
})
