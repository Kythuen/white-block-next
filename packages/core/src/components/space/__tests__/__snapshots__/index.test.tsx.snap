// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`# Space > ## Base > ### default config 1`] = `
<div
  class="wb-space wb-space--start wb-space--md"
  flex="~ row wrap"
  gap="$wb-space-gap"
  items="$wb-space-align"
>
  <!---->
</div>
`;
